driven/storage/adapter.go

```diff
@@ -3476,86 +3471,127 @@ func (sa *Adapter) UpdateAuthTypes(ID string, code string, description string, i
 
 // ============================== ServiceRegs ==============================
 
+// MigrateServiceRegs migrates all service registrations from the service_regs collection to the service_registrations collection
+// TODO: Remove this once all necessary migrations are complete
+func (sa *Adapter) MigrateServiceRegs() error {
+	transaction := func(context TransactionContext) error {
+		filter := bson.M{"core_host": sa.host}
+		count, err := sa.db.serviceRegistrations.CountDocumentsWithContext(context, filter)
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionFind, model.TypeServiceReg, &logutils.FieldArgs{"core_host": sa.host, "service_id": "all"}, err)
+		}
+
+		if count > 0 {
+			return nil
+		}
+
+		filter = bson.M{}
+		var results []model.ServiceRegistration
+		err = sa.db.serviceRegs.FindWithContext(context, filter, &results, nil)
+		if err != nil {
+			return errors.WrapErrorAction(logutils.ActionFind, model.TypeServiceReg, &logutils.FieldArgs{"core_host": sa.host, "service_id": "all"}, err)
```

Comment:
The error here should indicate that the service registrations found have `"core_host" != sa.host`.